internal/services/securitycenter/security_center_subscription_pricing_resource.go

```diff
@@ -79,11 +79,36 @@ func resourceSecurityCenterSubscriptionPricing() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
 			},
+			"extension": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"enabled": {
```

Comment:
In cases like these where there is a block that has an "enabled" property we tend to remove it to simplify the schema. Instead, we infer whether or not it is enabled based on if the block has been used. Would that be possible to do here? There is an example of this in the contributing guide [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md) (Example B).

Reply:
Hey,
I have consulted, and we want to keep it like this, since by default extension is enabled

Reply:
Thanks @Ihab-Zhaika - I have investigated this further with my colleagues and we believe removing `enabled` here  and inferring  whether or not an extension is enabled based on being present in the config is still possible in this case. In the case of users who wish to use the default values, we should be able to accommodate this by using d.HasChanges in the update function. This allows for specifying `ignore_changes` on `extensions` in the config, so that the default values are not overwritten. It would be good to add a note in the docs as well mentioning that we can use  `ignore_changes` here to retain the default settings.

Reply:
@Ihab-Zhaika

Whilst historically we've supported the `enabled` field, we're no longer supporting this for new functionality - and we're planning to remove most of the `enabled` fields in the next major version of the Provider, in favour of using the presence of a block to determine if the block is enabled/disabled.

In this case, this would mean that when a user specifies an extension in their Terraform Configuration, that the Extension should be enabled - and when it's not (i.e. the block is omitted for that Extension) then the Extension would be disabled. By using `d.HasChanges` above, we allow users who wish to use the default Extensions to do so by only performing updates when there's updates planned (which get filtered out via `d.HasChanges`).

As such, we'd need to make the following changes to make this behaviourally consistent with other parts of the Provider:

1. Remove the `enabled` field from this block
2. Update the `Read` method to only set any Enabled Extensions into the State
3. Update the `Update` method such that we disable any Extensions which aren't explicitly defined in the configuration - this can be done by retrieving the current state of the Subscription Pricing resource (and thus the Extensions already configured) - and then explicitly disabling any Extensions which aren't defined in the Configuration as a part of the Update Process). Since we're using `d.HasChanges`, this allows users who want to use the default Extensions to use `ignore_changes` on this field - which supports the functionality you're looking for in a slightly different manner.

Doing so will make this resource behaviourally consistent with other new functionality within the Provider (and as mentioned, we're intending to update the existing functionality in the next major version) - so if we can get that updated then we should be able to get this in 👍 

Thanks!

Reply:
@tombuildsstuff Done